blocks/api/factory.js

```diff
@@ -62,6 +67,44 @@ export function createBlock( name, blockAttributes = {} ) {
 }
 
 /**
+ * Returns an array of possible block transformations that could happen on the set of blocks received as argument.
+ *
+ * @param  {Array}  blocks Blocks array
+ * @return {Array}         Array of possible block transformations
+ */
+export function getPossibleBlockTransformations( blocks ) {
+	const sourceBlock = first( blocks );
+	if ( ! blocks || ! sourceBlock ) {
+		return [];
+	}
+	const isMultiBlock = blocks.length > 1;
+	const sourceBlockName = sourceBlock.name;
+
+	if ( isMultiBlock && ! every( blocks, { name: sourceBlockName } ) ) {
+		return [];
+	}
+
+	const blocksToBeTransformedFrom = filter( getBlockTypes(), type =>
+		find(
+			get( type, 'transforms.from', [] ),
+			t => t.type === 'block' && t.blocks.indexOf( sourceBlockName ) !== -1 &&
+			( ! isMultiBlock || t.isMultiBlock )
```

Comment:
It's hard to tell that this is part of the predicate for `find`, especially since the two lines above are separate arguments. I would suggest one of:

- Indent one more to indicate continuation of the condition
- Be more explicit with function arguments and avoid the arrow implicit return syntax

```js
const blocksToBeTransformedFrom = filter( getBlockTypes(), ( type ) => {
	return find( get( type, 'transforms.from', [] ), ( transform ) => {
		return (
			transform.type === 'block' &&
			transform.blocks.indexOf( sourceBlockName ) !== -1 &&
			( ! isMultiBlock || transform.isMultiBlock )
		);
	} );
} ).map( type => type.name );
```

(Note I also tend to discourage abbreviations. In this case it is very difficult to determine what `t` is - type? transform?)

- Split out separate named functions which better explain each step along the way

```js
const isTransformForBlockSource = ( sourceName ) => ( transform ) => (
	transform.type === 'block' &&
	transform.blocks.indexOf( sourceName ) !== -1 &&
	( ! isMultiBlock || transform.isMultiBlock )
);

const createIsTypeTransformableFrom = ( sourceName ) => ( type ) => (
	!! find(
		get( type, 'transforms.from', [] ),
		isTransformForBlockSource( sourceName ),
	)
);

const blocksToBeTransformedFrom = filter(
	getBlockTypes(),
	createIsTypeTransformableFrom( sourceBlockName ),
).map( type => type.name );
```

I think some inline comments (and/or named functions) would really go a long way to improve readability here. It's quite difficult to determine what's going on.